suite/suite_test.go

```diff
@@ -153,6 +154,31 @@ func (suite *SuiteTester) NonTestMethod() {
 	suite.NonTestMethodRunCount++
 }
 
+func (suite *SuiteTester) TestSubtest() {
+	beforeCount := suite.TestSubtestRunCount
+	suite.TestSubtestRunCount++
+
+	for _, t := range []struct {
+		testName string
+	}{
+		{"first"},
+		{"second"},
+	} {
+		suiteT := suite.T()
+		suite.Run(t.testName, func() {
+			// We should get a different *testing.T for subtests, so that
+			// go test recognizes them as proper subtests for output formatting
+			// and running individual subtests
+			subTestT := suite.T()
+			suite.NotEqual(subTestT, suiteT)
+		})
+		suite.Equal(suiteT, suite.T())
+	}
+
+	assert.Equal(suite.T(), suite.TestOneRunCount, beforeCount+1)
+	suite.Equal(suite.TestOneRunCount, beforeCount+1)
```

Comment:
These would seem to pass trivially since you capture and increment on lines 158 and 159. What am I missing?

Reply:
Its comparing `TestOneRunCount`, and the thing that got incremented was `TestSubtestRunCount`.

But as far as I can tell, that's an arbitrary test that was passing coincidentally.  Its been a few months since I wrote that, and maybe I had a good reason then, but I can't figure it out now at all.  So I pulled those lines of the test out, to avoid future confusion.  Good catch!